src/Microsoft.AspNetCore.Mvc.Razor.Extensions/RazorExtensions.cs

```diff
@@ -25,6 +25,8 @@ public static void Register(RazorProjectEngineBuilder builder)
             InheritsDirective.Register(builder);
             SectionDirective.Register(builder);
 
+            builder.Features.Add(new ViewComponentTagHelperDescriptorProvider());
```

Comment:
We should be adding this all of the time. The fact that this code is selective about when we add it creates all kinds of complications.